### DISCUSSION TITLE
Welcome to EmuFlight Discussions!

### DISCUSSION THREAD

**Discussion by nerdCopter** [emuflight/EmuFlight#477] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.
  * "_Discussions_" is not only new to GitHub, but new to EmuFlight; Please bare with us. Discord is still our primary community UI. (https://discord.gg/gdP9CwE)

## EmuFlight testers wanted!
* EmuFlight is always looking for test pilots to test development .hex firmware.
* In-trade for latest code (test builds), we ask that thorough testing and feedback be provided.
* Constructive feedback includes opinions whether pro or con and A/B analysis and blackbox logging and submission.
* Please be advised that most communications will be done in Discord, but we are willing to co-operate in GitHub Discussions as well.
* Please apply to be a test in Discord's `#testers-rollcall` channel.

## EmuFlight contributors wanted!
* We are a very small group of volunteers who need your help.
* We welcome any contributions in the form of code, information, or peer-assistance.
* Please consider making Pull-Requests for features, back-ports, targets, language translations, or fixes of any kind.
* Although not necessary, you are welcome to discuss in Discord's `#contributors` channel as well.
* We appreciate any help to peers in our regular Discord channels as well.

**Comment by chasingfpv** [comment]
i just flew a few packs on emuflight an look for tuning notes to help understand more .motors are a little warm i made a few little trys of trying a slight tune.like hoe emu feels from beta just need guideince. an i would test also have 8 quads a7 inch 3 5inch little whoops

**Reply by nerdCopter** [reply]
Welcome to EmuFlight!
I usually tell people to tune their dTerm LPF's first. defaults values may not be working well for you. Here is a most basic overview: https://github.com/emuflight/EmuFlight/wiki/LowPassFilters

That said, warm is perfectly fine, what you do not want is "HOT".  Do not allow your motors to get hot enough to where you cannot hold them for 12 seconds.

If you join the Discord, people could help with BlackBox Log  analysis as well.

**Comment by F101Voodoo** [comment]
Thank you for your Emuflight development. I installed a Walksnail Kit V2 on a 5” quad with heliospring FC. Used your Emuflight Configurator and flashed FC. Initial flights went well, good Betaflight OSD. Now OSD is freezing First noticed frozen data in flight at 3 minute mark. Then after cycling power to quad same frozen data persisted. After cycling power to both quad and Tx (ELRS) no BF OSD at all. Later at home powered up and good OSD data reappeared, but when I armed the quad it froze again instantly, timer 2 at 0 seconds.

**Reply by nerdCopter** [reply]
what git commit? (master?) 
what configurator commit?
set PAL for Walksnail?

**Comment by F101Voodoo** [comment]
Emuflight 0.4.1 configurator (unstable)
EMUF 0.4.2 HELIOSPRING target (unstable)
OSD Video Format HD

Should it be set to PAL?

**Reply by nerdCopter** [reply]
i believe so, this is what other Walksnail users have done for text alignment.
for HelioSPring I suggest UART3 for MSP-Display-port. (or specifically not UART1) (this might be why it freezes for you)
if you want MSP-Displayport be sure to use latest development builds:
I'm running HDZero (MSP-Displayport) on HelioSpring v1 without freezes.
using these two builds:
https://github.com/emuflight/dev-unstable/releases/tag/20240124.171109-hex
https://github.com/emuflight/dev-unstable/releases/tag/20240124.164604-cfg

**Comment by F101Voodoo** [comment]
Thanks I’ll switch to OSD PAL and see if it is better. HD looks good with
the small font but freezes too much. As long as I can place data on the
screen where I want it I’ll be happy.

**Comment by F101Voodoo** [comment]
Oh yeah of course I tried to use UART1 first. Then I realized that could
not be done. No biggie.

**Comment by nerdCopter** [comment]
okay, i dont understand the freezes. but i do not have prior reports of freezing, nor do i have Walksnail for testing.
you might consider a lower pid-loop, but i dont know for fact.  i set my pid denominator to 3 to fly at 10.67k pid/gyro loops.

**Comment by F101Voodoo** [comment]
PID's for OSD? Way over my head but thank you. I will see if I can get a
PAL format Emuflight OSD in my Walksnail goggles and fly without having the
Emuflight OSD data freeze in the goggles. Just to be clear, The Walksnail
video and data is fine and I can continue flying, it is just the
Emuflight FC data like battery voltage, flight time, LQI and RSSI etc. that
freezes.

**Reply by nerdCopter** [reply]
1) not PID's per se... "PID loop frequency in kilohertz". smaller value will allow for more time for processes to finish and use less CPU resources. 
2) The PAL thing was needed in older builds, but maybe not this newer build. i dont know as i cannot test, and pilots no longer providing feedback.  i encourage you to join Emuflight's Discord server for easier communications, but not 100% necessary if you prefer not.

**Comment by F101Voodoo** [comment]
Thanks, I joined github to send you my findings but had trouble navigating
it, to have a conversation. I will try Discord. I will let you know if PAL
solves the problem or if I can see how to change the PID loop frequency but
(like I suspect most of us modern flyers) I may just install a current FC.
I am more of a build and fly guy than a problem solver, LOL.

**Reply by nerdCopter** [reply]
:+1: 

try 
![image](https://github.com/emuflight/EmuFlight/assets/56646290/449c932f-210d-4fb2-82ca-47ca2a514bb4)

or try
![image](https://github.com/emuflight/EmuFlight/assets/56646290/d6f3315f-1503-4e7a-9957-fa0e5e42ae0c)

**Comment by F101Voodoo** [comment]
I will, I'll let you know how it works out.

**Reply by nerdCopter** [reply]
i found a bug when i merged HD to master (but may have only affected analog, unsure). master is fixed. i will build new hex, watch dev-unstable repo. will take a couple hours.

**Comment by F101Voodoo** [comment]
Awesome, thanks for the update!

**Comment by F101Voodoo** [comment]
FYI I tried the PID settings you gave me but with all three Configuration
PIDs the OSD is intermittent. Im starting to think the Heliospring FC or
Walksnail Kit may be to blame. The Walksnail is a new unit. Ill try your
new brew but it might be your work is fine and something else is the
problem.
Cheers!

**Reply by nerdCopter** [reply]
Another pilot reports he cannot enter into OSD-menu with stick-command on WlakSnail. Are you able?

**Comment by F101Voodoo** [comment]
I never used stick commands with Walksnail. Don't know how... I only ever
used them with analog vtx smart audio. FYI I have not tried your newest
software revision yet, been busy. I'll let you know if it works better (no
frozen OSD). Thanks!

**Comment by F101Voodoo** [comment]
Hey I wanted you to know I have started having the same video problems with
my other Walksnail equipped quads which are using current versions of
BetaFlight. Namely frozen OSD data and even frozen video (crashing!) it is
a bummer and I will have to send my Walksnail GogglesX back in for service.
I have to assume that your programming to allow HD OSD with Emuflight is
fine. The goggles appear to be at fault.

**Reply by nerdCopter** [reply]
thank you very much for reporting this. sorry for your troubles.
